BraveShared/Analytics/DAU.swift

```diff
@@ -72,27 +84,34 @@ public class DAU {
     func paramsAndPrefsSetup() -> [URLQueryItem]? {
         var params = [channelParam(), versionParam()]
         
-        /// This is not the same as `firstLaunch` concept, due to DAU delay, this may var be `true` on a subsequent launch, if server ping failed
+        /// This is not the same as `firstLaunch` concept, due to DAU delay,
+        /// this may var be `true` on a subsequent launch, if server ping failed.
         let firstPing = Preferences.DAU.firstPingSuccess.value
+        if !firstPing {
+            log.debug("server ping was not successful yet")
+            return nil
+        }
+        
+        let firstLaunch = firstPing && Preferences.DAU.lastLaunchInfo.value == nil
```

Comment:
Please double check that this logic is correct now. It had a bug before, now it passes all unit tests.

Reply:
This would return `false` even if the first ping never occurred. `first=true` should continue to be attempted until it succeeds.

Reply:
It won't get there because few lines above we have:

```
if !firstPing {
            log.debug("server ping was not successful yet")
            return nil
        }
```

When do we set `firstPingSuccess` to false then? For next pings we would have to set it to false somewhere

Reply:
No, it is only for the firstPingEver, it is just flipped to `true` once we have communicated with the server the first time, then we never do again. It is not first ping this launch.